### DISCUSSION TITLE
How to remove pgsodium from an exist DB

### DISCUSSION THREAD

**Discussion by Aging-Developer** [supabase/supabase#23425] [Questions]
Hi, I have an exist DB in Supabase that was developed last year, and have a number of tables encrypted with pgsodium, and views over them, the solution isnt live yet (in 1-2 months time), after reading this: 
![image](https://github.com/supabase/supabase/assets/122821259/e4ea9d46-2424-4e95-a764-5fce6d962562)

I am thinking there is no way I want that pain when gone live and active users on the system, so best to remove it now!

I think the steps are:

1.  Remove the triggers used by encryption on the tables
2. drop views over the tables
3. remove the below from tables:

    key_id uuid not null default 'the key id'::uuid,
    nonce bytea null default pgsodium.crypto_aead_det_noncegen (),

4. Disable the extension (i doubt that would trigger points 1-3 automatically?)
5. remove the vault key
6. Hope that things work afterwards

Any thoughts on a better way to remove pgsodium?  and any snags if someone has already tried?

and any other cleaning up I would need to do?

**Comment by bombillazo** [comment]
We're also interested in understanding the alternative or path forward if we want to have encrypted data at the column level. We use the Vault plus custom encrypted tables/columns, but we're unsure how to proceed now...
